test/run_test.py

```diff
@@ -912,6 +996,18 @@ def exclude_tests(exclude_list, selected_tests, exclude_message=None):
     return selected_tests
 
 
+def must_serial(file: str) -> bool:
```

Comment:
Oh man, this looks super complicated, but well it's complicated

Reply:
its mostly just the more unique test configs that im hesitant about running in parallel